src/platform/plugins/shared/esql/public/triggers/esql_controls/control_flyout/value_control_form.tsx

```diff
@@ -218,13 +223,22 @@ export function ValueControlForm({
       if (initialState?.esqlQuery) {
         onValuesQuerySubmit(initialState.esqlQuery);
       } else if (valuesRetrieval) {
-        const queryForValues = `FROM ${getIndexPatternFromESQLQuery(
-          queryString
-        )} | STATS BY ${valuesRetrieval}`;
-        onValuesQuerySubmit(queryForValues);
+        const indexPattern = getIndexPatternFromESQLQuery(queryString);
+        const encodedQuery = encodeURIComponent(`FROM ${indexPattern}`);
+        core.http
```

Comment:
@momovdg as you haven't merged it yet, I was thinking that we should:

1. Have a catch too
2. I think we should await

So something like:

```
const response = (await http?.get(`${TIMEFIELD_ROUTE}${encodedQuery}`).catch((error) => {
    // eslint-disable-next-line no-console
    console.error('Failed to fetch the timefield', error);
    return undefined;
  })) as { timeField?: string } | undefined;
const timeField = response?.timeField;
```